internal/component/faro/receiver/receiver.go

```diff
@@ -243,3 +256,62 @@ func (vs *varSourceMapsStore) SetInner(inner sourceMapsStore) {
 
 	vs.inner = inner
 }
+
+func (c *Component) stopCleanup() {
+	c.cleanupMut.Lock()
+	defer c.cleanupMut.Unlock()
+	if c.cleanup != nil {
+		c.cleanup.cancel()  // signal goroutines to exit
+		c.cleanup.wg.Wait() // wait for them
+		c.cleanup = nil
+	}
+}
+
+func (c *Component) startCleanup(args Arguments, s *sourceMapsStoreImpl) {
```

Comment:
This should be part of source map store instead. It could have Start and Stop methods. Then we don't need an additional mutex